reactor-netty-http/src/main/java/reactor/netty/http/server/HttpCompressionSettingsSpec.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2020-2023 VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.netty.handler.codec.compression.Brotli;
+import io.netty.handler.codec.compression.BrotliOptions;
+import io.netty.handler.codec.compression.CompressionOptions;
+import io.netty.handler.codec.compression.DeflateOptions;
+import io.netty.handler.codec.compression.GzipOptions;
+import io.netty.handler.codec.compression.SnappyOptions;
+import io.netty.handler.codec.compression.StandardCompressionOptions;
+import io.netty.handler.codec.compression.Zstd;
+import io.netty.handler.codec.compression.ZstdOptions;
+import io.netty.util.internal.ObjectUtil;
+
+/**
+ * HTTP Compression configuration builder for the {@link SimpleCompressionHandler}.
+ *
+ * @author raccoonback
+ */
+public final class HttpCompressionSettingsSpec {
+
+	private final GzipOptions gzipOptions;
+	private final DeflateOptions deflateOptions;
+	private final SnappyOptions snappyOptions;
+	private BrotliOptions brotliOptions;
+	private ZstdOptions zstdOptions;
+
+	private HttpCompressionSettingsSpec() {
+		gzipOptions = StandardCompressionOptions.gzip();
+		deflateOptions = StandardCompressionOptions.deflate();
+		snappyOptions = StandardCompressionOptions.snappy();
+
+		if (Brotli.isAvailable()) {
+			brotliOptions = StandardCompressionOptions.brotli();
+		}
+
+		if (Zstd.isAvailable()) {
+			zstdOptions = StandardCompressionOptions.zstd();
+		}
+	}
```

Comment:
The `default constructor` sets the available compression to **default settings**.